src/vulkan-renderer/imgui/imgui.cpp

```diff
@@ -0,0 +1,616 @@
+#include "inexor/vulkan-renderer/imgui/imgui.hpp"
+#include "inexor/vulkan-renderer/wrapper/make_info.hpp"
+
+#include <cassert>
+#include <stdexcept>
+
+namespace inexor::vulkan_renderer::imgui {
+
+ImGUIOverlay::ImGUIOverlay(ImGUIOverlay &&other) noexcept
```

Comment:
> Clang-Tidy: Constructor does not initialize these fields: pushConstBlock
Clang-Tidy: Constructor does not initialize these fields: pushConstBlock
Clang-Tidy: Move constructor initializes class member by calling a copy constructor
Clang-Tidy: Move constructor initializes class member by calling a copy constructor****

Reply:
I initialize the push constant block in the constructor now:

```cpp
// Initializ push constant block
m_push_const_block.scale = glm::vec2(0.0f, 0.0f);
m_push_const_block.translate = glm::vec2(0.0f, 0.0f);

```

Reply:
I also added this to the move constructor:

```cpp
m_push_const_block(other.m_push_const_block)
```

Done.